src/systems/physics/Physics.cc

```diff
@@ -2246,31 +2282,43 @@ void PhysicsPrivate::UpdateCollisions(EntityComponentManager &_ecm)
       [&](const Entity &_collEntity1, components::Collision *,
           components::ContactSensorData *_contacts) -> bool
       {
+        msgs::Contacts contactsComp;
         if (entityContactMap.find(_collEntity1) == entityContactMap.end())
         {
           // Clear the last contact data
-          *_contacts = components::ContactSensorData();
+          auto state = _contacts->SetData(contactsComp,
+            this->contactsEql) ?
+            ComponentState::OneTimeChange :
+            ComponentState::NoChange;
+          _ecm.SetChanged(
+            _collEntity1, components::ContactSensorData::typeId, state);
           return true;
         }
-
-        const auto &contactMap = entityContactMap[_collEntity1];
-
-        msgs::Contacts contactsComp;
-
-        for (const auto &[collEntity2, contactData] : contactMap)
+        else
```

Comment:
I don't think this block needs to be inside an `else`, because the `if` above is returning. We could save some indentation by removing the `else`.

Alternatively, you could remove the block that's inside the `if` and let the `SetData` below set the empty `contactsComp`. Makes sense?